src/providers/mbed_provider/key_management.rs

```diff
@@ -46,42 +45,41 @@ fn create_key_id(
     key_triple: KeyTriple,
     key_attributes: Attributes,
     store_handle: &mut dyn ManageKeyInfo,
-    local_ids_handle: &mut LocalIdStore,
+    max_current_id: &AtomicU32,
 ) -> Result<key::psa_key_id_t> {
-    let mut rng = SmallRng::from_entropy();
-    let mut key_id = rng.gen_range(key::PSA_KEY_ID_USER_MIN, key::PSA_KEY_ID_USER_MAX + 1);
-
-    while local_ids_handle.contains(&key_id) {
-        key_id = rng.gen_range(key::PSA_KEY_ID_USER_MIN, key::PSA_KEY_ID_USER_MAX + 1);
+    // fetch_add adds 1 to the old value and returns the old value, so add 1 for new ID
+    let new_key_id = max_current_id.fetch_add(1, Relaxed) + 1;
+    if new_key_id > key::PSA_KEY_ID_USER_MAX {
+        // If storing key failed and no other keys were created in the mean time, it is safe to
+        // decrement the key counter.
+        let _ = max_current_id.compare_and_swap(new_key_id, new_key_id - 1, Relaxed);
+        return Err(ResponseStatus::PsaErrorBadState);
```

Comment:
Wasn't sure on the error to return if there are no more key IDs available.

Reply:
I think `BadState` is an error referring to multi-parts operations. It might be better to use `InsufficientMemory`. Also, could you please add an `error!` log to note that we've reached the limit?

Reply:
Is it a good idea to return `InsufficientMemory`? I guess in conjuction with a log message it would be easy to understand what's going on, but based on the error code alone it might be confusing.

Reply:
I was thinking about adding a new error but didn't want to touch `parsec-interface` without discussing first.

Reply:
I think the problem with a new response code like that is we shouldn't be exposing internal issues (or, not ones that mean nothing to a client). `InsufficientMemory` is probably ok if we document what it means in our context - we just add an entry [here](https://parallaxsecond.github.io/parsec-book/parsec_client/operations/psa_generate_key.html#specific-response-status-codes)